### ISSUE TITLE
At  Typesense Delete all the collection at Kubectl

### ISSUE THREAD

**Issue by arn-ob** [typesense/typesense#1125]
## Description

Here I deploy the collection, But after some days deletes everything.  How I solve this issue 

**Typesense Version**:
v0.25.0.rc45
**OS**:
kubectl v1.26.2-eks-a59e1f0

**Comment by donchev7** [comment]
We are experiencing a similar issue. Collection that have been previously been created are suddenly gone.

The server returns 404 errors because of it.

**Comment by jasonbosco** [comment]
This usually means that all or majority of nodes in the cluster have somehow been rotated by k8s at the same time, and so the cluster has lost quorum. 

For a 3 node cluster, you want to ensure that the cluster does not suffer a loss of more than 1 node at any given time (this is how the RAFT consensus algorithm works). 

If two nodes are lost in the cluster, then you'd need to reset the cluster state and bring up the 2 nodes back up again as described here: https://typesense.org/docs/guide/high-availability.html#recovering-a-cluster-that-has-lost-quorum

**Comment by donchev7** [comment]
Yeah sure but logs do not indicate that the cluster lost a quorum.
Also no restarts happened to the pods the last 23 days...

**Comment by donchev7** [comment]
Ok I see. It can somehow miss a quorum without a restart. This is what I just witnessed now:

```
➜ k -n typesense get po
NAME          READY   STATUS    RESTARTS   AGE
typesense-0   0/1     Running   0          20d
typesense-1   0/1     Running   0          20d
typesense-2   1/1     Running   0          20d

**Comment by jasonbosco** [comment]
Could you share the last 50 lines from each of the nodes, just before this happens?

**Comment by donchev7** [comment]
I converted the cluster to single node. Even having a single node the collections are getting deleted! Is this expected behaviour?

```
I20240702 12:35:52.528146   262 raft_server.cpp:693] Term: 4, pending_queue: 0, last_index: 1631, committed: 1631, known_applied: 1631, applying: 0, pending_writes: 0, queued_writes: 0, local_sequence: 9712 |  
  |   | 2024-07-02 16:35:43.258 | I20240702 12:35:43.257954   291 raft_server.h:276] Node becomes leader, term: 4 |  
  |   | 2024-07-02 16:35:43.258 | I20240702 12:35:43.257928   291 node.cpp:3298] node default_group:10.244.5.63:8107:8108 reset ConfigurationCtx, new_peers: 10.244.5.63:8107:8108, old_peers: 10.244.5.63:8107:8108 |  
  |   | 2024-07-02 16:35:43.258 | I20240702 12:35:43.257853   291 raft_server.h:293] Configuration of this group is 10.244.5.63:8107:8108 |  
  |   | 2024-07-02 16:35:43.210 | W20240702 12:35:43.210091   286 field.cpp:17] Collection schema cannot contain a field with name `id`. Ignoring field. |  
  |   | 2024-07-02 16:35:43.200 | I20240702 12:35:43.200605   291 node.cpp:1899] node default_group:10.244.5.63:8107:8108 term 4 become leader of group 10.244.5.63:8107:8108 |  
  |   | 2024-07-02 16:35:43.200 | I20240702 12:35:43.200527   291 raft_meta.cpp:546] Saved single stable meta, path /data/state/meta term 4 votedfor 10.244.5.63:8107:8108 time: 10296 |  
  |   | 2024-07-02 16:35:43.190 | I20240702 12:35:43.190135   291 node.cpp:1645] node default_group:10.244.5.63:8107:8108 term 3 start vote and grant vote self |  
  |   | 2024-07-02 16:35:43.190 | I20240702 12:35:43.189987   291 node.cpp:1579] node default_group:10.244.5.63:8107:8108 term 3 start pre_vote |  
  |   | 2024-07-02 16:35:42.527 | W20240702 12:35:42.526993   262 raft_server.cpp:718] Node with no leader. Resetting peers of size: 1 |  
  |   | 2024-07-02 16:35:42.527 | I20240702 12:35:42.526924   262 raft_server.cpp:693] Term: 3, pending_queue: 0, last_index: 1630, committed: 0, known_applied: 1385, applying: 0, pending_writes: 0, queued_writes: 0, local_sequence: 8490 |  
  |   | 2024-07-02 16:35:32.526 | I20240702 12:35:32.525882   262 raft_meta.cpp:546] Saved single stable meta, path /data/state/meta term 3 votedfor 0.0.0.0:0:0 time: 13222 |  
  |   | 2024-07-02 16:35:32.512 | W20240702 12:35:32.512624   262 node.cpp:926] node default_group:10.244.5.63:8107:8108 set_peer from 10.244.5.59:8107:8108 to 10.244.5.63:8107:8108 |  
  |   | 2024-07-02 16:35:32.512 | W20240702 12:35:32.512611   262 raft_server.cpp:718] Node with no leader. Resetting peers of size: 1 |  
  |   | 2024-07-02 16:35:32.512 | I20240702 12:35:32.512594   262 raft_server.cpp:693] Term: 2, pending_queue: 0, last_index: 1630, committed: 0, known_applied: 1385, applying: 0, pending_writes: 0, queued_writes: 0, local_sequence: 8490 |  
  |   | 2024-07-02 16:35:32.512 | W20240702 12:35:32.512575   262 controller.cpp:1550] SIGINT was installed with 1 |  
  |   | 2024-07-02 16:35:32.512 | I20240702 12:35:32.512565   262 typesense_server_utils.cpp:293] Snapshot max byte count configured as: 4194304 |  
  |   | 2024-07-02 16:35:32.512 | I20240702 12:35:32.512559   262 typesense_server_utils.cpp:292] Snapshot interval configured as: 3600s |  
  |   | 2024-07-02 16:35:32.512 | I20240702 12:35:32.512552   262 typesense_server_utils.cpp:291] Typesense peering service is running on 10.244.5.63:8107 |  
  |   | 2024-07-02 16:35:32.512 | I20240702 12:35:32.512538   262 raft_server.cpp:134] Node last_index: 1630 |  
  |   | 2024-07-02 16:35:32.512 | I20240702 12:35:32.512499   262 node.cpp:608] node default_group:10.244.5.63:8107:8108 init, term: 2 last_log_id: (index=1630,term=2) conf: 10.244.5.59:8107:8108 old_conf: |  
  |   | 2024-07-02 16:35:32.512 | I20240702 12:35:32.512460   262 raft_meta.cpp:521] Loaded single stable meta, path /data/state/meta term 2 votedfor 10.244.5.59:8107:8108 time: 210 |  
  |   | 2024-07-02 16:35:32.512 | I20240702 12:35:32.512099   291 snapshot_executor.cpp:264] node default_group:10.244.5.63:8107:8108 snapshot_load_done, last_included_index: 1385 last_included_term: 2 peers: "10.244.5.59:8107:8108" |  
  |   | 2024-07-02 16:35:32.511 | I20240702 12:35:32.511622   291 raft_server.h:293] Configuration of this group is 10.244.5.59:8107:8108 |  
  |   | 2024-07-02 16:35:32.511 | I20240702 12:35:32.511587   291 batched_indexer.cpp:617] Restored 0 in-flight requests from snapshot. |  
  |   | 2024-07-02 16:35:32.511 | I20240702 12:35:32.511555   291 raft_server.cpp:630] Initializing batched indexer from snapshot state... |  
  |   | 2024-07-02 16:35:32.511 | I20240702 12:35:32.511528   291 raft_server.cpp:623] Finished loading collections from disk. |  
  |   | 2024-07-02 16:35:32.510 | I20240702 12:35:32.510476   291 collection_manager.cpp:425] Loaded 2 collection(s). |  
  |   | 2024-07-02 16:35:32.510 | I20240702 12:35:32.510413   358 collection_manager.cpp:360] Loaded 2 collection(s) so far |  
  |   | 2024-07-02 16:35:32.510 | I20240702 12:35:32.510351   357 collection_manager.cpp:360] Loaded 1 collection(s) so far |  
  |   | 2024-07-02 16:35:32.510 | I20240702 12:35:32.510339   358 collection_manager.cpp:2276] Indexed 60/60 documents into collection alpha_entities |  
  |   | 2024-07-02 16:35:32.510 | I20240702 12:35:32.510305   357 collection_manager.cpp:2276] Indexed 22/22 documents into collection alpha_accounts |  
  |   | 2024-07-02 16:35:32.509 | I20240702 12:35:32.509279   358 collection_manager.cpp:2158] Loading collection alpha_entities |  
  |   | 2024-07-02 16:35:32.509 | I20240702 12:35:32.509266   357 collection_manager.cpp:2158] Loading collection alpha_accounts |  
  |   | 2024-07-02 16:35:32.509 | I20240702 12:35:32.509181   358 collection_manager.cpp:160] Found collection alpha_entities with 4 memory shards. |  
  |   | 2024-07-02 16:35:32.509 | I20240702 12:35:32.509011   357 collection_manager.cpp:160] Found collection alpha_accounts with 4 memory shards. |  
  |   | 2024-07-02 16:35:32.507 | I20240702 12:35:32.507210   291 collection_manager.cpp:308] Found 2 collection(s) on disk. |  
  |   | 2024-07-02 16:35:32.507 | I20240702 12:35:32.507179   291 collection_manager.cpp:299] Loading upto 32 collections in parallel, 1000 documents at a time. |  
  |   | 2024-07-02 16:35:32.507 | I20240702 12:35:32.507144   291 auth_manager.cpp:34] Indexing 0 API key(s) found on disk. |  
  |   | 2024-07-02 16:35:32.507 | I20240702 12:35:32.507052   291 collection_manager.cpp:263] CollectionManager::load() |  
  |   | 2024-07-02 16:35:32.507 | I20240702 12:35:32.507042   291 raft_server.cpp:616] Loading collections from disk... |  
  |   | 2024-07-02 16:35:32.507 | I20240702 12:35:32.507006   291 store.h:323] DB open success! |  
  |   | 2024-07-02 16:35:32.446 | I20240702 12:35:32.446621   291 store.h:64] Initializing DB by opening state dir: /data/db |  
  |   | 2024-07-02 16:35:32.446 | I20240702 12:35:32.446588   291 store.h:309] copy snapshot /data/state/snapshot/snapshot_00000000000000001385/db_snapshot to /data/db success |  
  |   | 2024-07-02 16:35:32.446 | I20240702 12:35:32.445981   291 store.h:299] rm /data/db success |  
  |   | 2024-07-02 16:35:32.444 | I20240702 12:35:32.444459   291 raft_server.cpp:647] on_snapshot_load |  
  |   | 2024-07-02 16:35:32.431 | I20240702 12:35:32.431386   262 log.cpp:1112] load open segment, path: /data/state/log first_index: 1 |  
  |   | 2024-07-02 16:35:32.431 | I20240702 12:35:32.431322   262 log.cpp:1172] log load_meta /data/state/log/log_meta first_log_index: 1334 time: 294 |  
  |   | 2024-07-02 16:35:32.431 | I20240702 12:35:32.431002   262 log.cpp:690] Use murmurhash32 as the checksum type of appending entries |  
  |   | 2024-07-02 16:35:32.430 | I20240702 12:35:32.430701   262 raft_server.cpp:68] Nodes configuration: 10.244.5.63:8107:8108 |  
  |   | 2024-07-02 16:35:32.430 | I20240702 12:35:32.430346   262 server.cpp:1184] Check out http://typesense-0:8107 in web browser. |  
  |   | 2024-07-02 16:35:32.430 | I20240702 12:35:32.430305   262 server.cpp:1181] Server[braft::RaftStatImpl+braft::FileServiceImpl+braft::RaftServiceImpl+braft::CliServiceImpl] is serving on port=8107. |  
  |   | 2024-07-02 16:35:32.426 | I20240702 12:35:32.426403   263 batched_indexer.cpp:188] BatchedIndexer skip_index: -9999 |  
  |   | 2024-07-02 16:35:32.414 | I20240702 12:35:32.414445   265 typesense_server_utils.cpp:490] Conversation garbage collector thread started. |  
  |   | 2024-07-02 16:35:32.413 | I20240702 12:35:32.413316   262 typesense_server_utils.cpp:243] Since no --nodes argument is provided, starting a single node Typesense cluster. |  
  |   | 2024-07-02 16:35:32.411 | I20240702 12:35:32.411195   263 batched_indexer.cpp:183] Starting batch indexer with 64 threads. |  
  |   | 2024-07-02 16:35:32.411 | I20240702 12:35:32.411109     1 http_server.cpp:178] Typesense has started listening on port 8108 |  
  |   | 2024-07-02 16:35:32.411 | I20240702 12:35:32.410861     1 typesense_server_utils.cpp:547] Starting API service... |  
  |   | 2024-07-02 16:35:32.410 | I20240702 12:35:32.410218     1 typesense_server_utils.cpp:466] Loaded 0(s) conversation models. |  
  |   | 2024-07-02 16:35:32.410 | I20240702 12:35:32.410200     1 typesense_server_utils.cpp:458] Loaded 0(s) conversations. |  
  |   | 2024-07-02 16:35:32.410 | I20240702 12:35:32.410133     1 ratelimit_manager.cpp:547] Loaded 0 rate limit bans. |  
  |   | 2024-07-02 16:35:32.410 | I20240702 12:35:32.410099     1 ratelimit_manager.cpp:546] Loaded 0 rate limit rules. |  
  |   | 2024-07-02 16:35:32.344 | I20240702 12:35:32.344624     1 store.h:64] Initializing DB by opening state dir: /data/meta |  
  |   | 2024-07-02 16:35:32.284 | I20240702 12:35:32.284267     1 store.h:64] Initializing DB by opening state dir: /data/db |  
  |   | 2024-07-02 16:35:32.257 | I20240702 12:35:32.257357     1 typesense_server_utils.cpp:395] Thread pool size: 64 |  
  |   | 2024-07-02 16:35:32.256 | I20240702 12:35:32.256609     1 typesense_server_utils.cpp:344] Typesense is using jemalloc. |  
  |   | 2024-07-02 16:35:32.256 | I20240702 12:35:32.256559     1 typesense_server_utils.cpp:341] Starting Typesense 26.0 |  
  |   | 2024-07-02 16:35:29.967 | I20240702 12:35:29.967106     1 typesense_server_utils.cpp:577] Bye. |  
  |   | 2024-07-02 16:35:29.926 | I20240702 12:35:29.926425     1 typesense_server_utils.cpp:570] CollectionManager dispose, this might take some time... |  
  |   | 2024-07-02 16:35:29.926 | I20240702 12:35:29.926223     1 typesense_server_utils.cpp:566] Deleting server |  
  |   | 2024-07-02 16:35:29.926 | I20240702 12:35:29.926213     1 typesense_server_utils.cpp:562] CURL clean up |  
  |   | 2024-07-02 16:35:29.926 | I20240702 12:35:29.926175     1 typesense_server_utils.cpp:558] Deleting batch indexer |  
  |   | 2024-07-02 16:35:29.926 | I20240702 12:35:29.926019     1 typesense_server_utils.cpp:554] Typesense API service has quit. |  
  |   | 2024-07-02 16:35:29.924 | I20240702 12:35:29.924285   262 typesense_server_utils.cpp:541] Shutting down replication_thread_pool. |  
  |   | 2024-07-02 16:35:29.921 | I20240702 12:35:29.921542   262 typesense_server_utils.cpp:537] Shutting down app_thread_pool. |  
  |   | 2024-07-02 16:35:29.919 | I20240702 12:35:29.918846   262 typesense_server_utils.cpp:533] Shutting down server_thread_pool |  
  |   | 2024-07-02 16:35:29.918 | I20240702 12:35:29.918692   262 typesense_server_utils.cpp:529] Waiting for housekeeping thread to be done... |  
  |   | 2024-07-02 16:35:29.918 | I20240702 12:35:29.918603   262 typesense_server_utils.cpp:526] Waiting for conversation garbage collector thread to be done... |  
  |   | 2024-07-02 16:35:29.918 | I20240702 12:35:29.918569   262 typesense_server_utils.cpp:523] Shutting down conversation garbage collector thread... |  
  |   | 2024-07-02 16:35:29.918 | I20240702 12:35:29.918471   262 typesense_server_utils.cpp:520] Waiting for event sink thread to be done... |  
  |   | 2024-07-02 16:35:29.918 | I20240702 12:35:29.918426   262 typesense_server_utils.cpp:517] Shutting down event sink thread... |  
  |   | 2024-07-02 16:35:29.916 | I20240702 12:35:29.916618   263 batched_indexer.cpp:478] Batched indexer threadpool shutdown... |  
  |   | 2024-07-02 16:35:29.916 | I20240702 12:35:29.916297   263 batched_indexer.cpp:474] Notifying reference sequence thread about shutdown... |  
  |   | 2024-07-02 16:35:29.915 | I20240702 12:35:29.915711   263 batched_indexer.cpp:468] Notifying batch indexer threads about shutdown... |  
  |   | 2024-07-02 16:35:29.512 | I20240702 12:35:29.512297   262 typesense_server_utils.cpp:514] Waiting for batch indexing thread to be done... |  
  |   | 2024-07-02 16:35:29.512 | I20240702 12:35:29.512266   262 typesense_server_utils.cpp:511] Shutting down batch indexer... |  
  |   | 2024-07-02 16:35:29.511 | I20240702 12:35:29.511008   262 typesense_server_utils.cpp:335] Typesense peering service has quit. |  
  |   | 2024-07-02 16:35:29.510 | I20240702 12:35:29.510911   262 typesense_server_utils.cpp:332] raft_server.join() |  
  |   | 2024-07-02 16:35:29.510 | I20240702 12:35:29.510517   262 server.cpp:1241] Server[braft::RaftStatImpl+braft::FileServiceImpl+braft::RaftServiceImpl+braft::CliServiceImpl] is going to quit |  
  |   | 2024-07-02 16:35:29.510 | I20240702 12:35:29.510486   262 typesense_server_utils.cpp:329] raft_server.stop() |  
  |   | 2024-07-02 16:35:29.510 | I20240702 12:35:29.510293   262 node.cpp:961] node default_group:10.244.5.59:8107:8108 shutdown, current_term 2 state SHUTDOWN |  
  |   | 2024-07-02 16:35:29.510 | I20240702 12:35:29.510231   295 raft_server.h:285] This node is down |  
  |   | 2024-07-02 16:35:29.510 | I20240702 12:35:29.510200   262 raft_server.cpp:991] node->join |  
  |   | 2024-07-02 16:35:29.510 | I20240702 12:35:29.509753   262 replicator.cpp:1499] Group default_group Fail to find the next candidate |  
  |   | 2024-07-02 16:35:29.509 | I20240702 12:35:29.509791   295 raft_server.h:281] Node stepped down : Raft node is going to quit. |  
  |   | 2024-07-02 16:35:29.509 | I20240702 12:35:29.509662   262 node.cpp:961] node default_group:10.244.5.59:8107:8108 shutdown, current_term 2 state LEADER |  
  |   | 2024-07-02 16:35:29.509 | I20240702 12:35:29.509645   262 raft_server.cpp:987] node->shutdown |  
  |   | 2024-07-02 16:35:29.509 | I20240702 12:35:29.509231   262 raft_server.cpp:983] Replication state shutdown, store sequence: 9712 |  
  |   | 2024-07-02 16:35:29.509 | I20240702 12:35:29.509225   262 raft_server.cpp:977] Waiting for in-flight writes to finish... |  
  |   | 2024-07-02 16:35:29.509 | I20240702 12:35:29.509217   262 raft_server.cpp:973] Set shutting_down = true |  
  |   | 2024-07-02 16:35:29.509 | I20240702 12:35:29.509150   262 typesense_server_utils.cpp:324] Typesense peering service is going to quit. |  
  |   | 2024-07-02 16:35:28.782 | I20240702 12:35:28.782492     1 typesense_server_utils.cpp:58] Stopping Typesense server... |  
  |   | 2024-07-02 16:35:19.508 | I20240702 12:35:19.508038   295 raft_server.h:60] Peer refresh succeeded!
```

**Comment by jasonbosco** [comment]
> is this the problem why the collection is not being restored?
> 
> ```
> Collection schema cannot contain a field with name `id`
> ```

No, that shouldn't be an issue. It's just a warning. 

The logs you posted seem to be from multiple nodes running at the same time. So it's hard to parse what's going on. 

Could you post the logs just for a single node, from the time it starts until you see at least 3 or more logs of "Peer refresh succeeded"?

**Comment by donchev7** [comment]
You were right. It does not have anything todo with the `id`. We narrowed it down to doing frequent (every minute) **client.import**  calls. After we switched to **client.create(news, { action: 'emplace' })** it seems to be fine
